pkg/cloud/amazon/vault/vault_resources.go

```diff
@@ -0,0 +1,374 @@
+	iamUser := resources.AWSIAMUser{
+		UserName: username,
+	}
+	return iamUser
+}
+
+func createIamUserPolicy(username string, depends []string) resources.AWSIAMPolicy {
+	type PolicyDocument struct {
+		Sid      string
+		Effect   string
+		Action   []string
+		Resource interface{}
+	}
+
+	type PolicyRoot struct {
+		Version   string
+		Statement []PolicyDocument
+	}
+
+	policyName := "vault_" + username
+
+	policyDocument := resources.AWSIAMPolicy{
+		PolicyName: policyName,
+		Users:      []string{username},
+		PolicyDocument: PolicyRoot{
+			Version: "2012-10-17",
+			Statement: []PolicyDocument{
+				{
+					Sid:    "DynamoDB",
+					Effect: "Allow",
+					Action: []string{
+						"dynamodb:DescribeLimits",
+						"dynamodb:DescribeTimeToLive",
+						"dynamodb:ListTagsOfResource",
+						"dynamodb:DescribeReservedCapacityOfferings",
+						"dynamodb:DescribeReservedCapacity",
+						"dynamodb:ListTables",
+						"dynamodb:BatchGetItem",
+						"dynamodb:BatchWriteItem",
+						"dynamodb:CreateTable",
+						"dynamodb:DeleteItem",
+						"dynamodb:GetItem",
+						"dynamodb:GetRecords",
+						"dynamodb:PutItem",
+						"dynamodb:Query",
+						"dynamodb:UpdateItem",
+						"dynamodb:Scan",
+						"dynamodb:DescribeTable",
+					},
+					Resource: "DYNAMODB_JSON",
+				},
+				{
+					Sid:    "S3",
+					Effect: "Allow",
+					Action: []string{
+						"s3:PutObject",
+						"s3:GetObject",
+					},
+					Resource: "BUCKET_JSON",
+				},
+				{
+					Sid:    "S3List",
+					Effect: "Allow",
+					Action: []string{
+						"s3:ListBucket",
+					},
+					Resource: "BUCKET_LIST_JSON",
+				},
+				{
+					Sid:    "KMS",
+					Effect: "Allow",
+					Action: []string{
+						"kms:Encrypt",
+						"kms:Decrypt",
+					},
+					Resource: "KMS_SUB_JSON",
+				},
+			},
+		},
+	}
+
+	policyDocument.SetDependsOn(depends)
+
+	return policyDocument
+}
+
+func runCloudformationTemplate(templateBody *string, stackName *string) error {
+
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	// Create CloudFormation client in region
+	svc := cloudformation.New(sess)
+
+	desInput := &cloudformation.DescribeStacksInput{StackName: stackName}
+
+	input := &cloudformation.CreateStackInput{
+		TemplateBody: templateBody,
+		StackName:    stackName,
+		Capabilities: []*string{
+			aws.String("CAPABILITY_NAMED_IAM"),
+		},
+	}
+
+	_, err := svc.CreateStack(input)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to create vault preresiquite resources")
+	}
+
+	// Wait until stack is created
+	err = svc.WaitUntilStackCreateComplete(desInput)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to create vault preresiquite resources")
+	}
+
+	log.Logger().Infof("Ran the vault cloudformation template successfully")
+	return nil
+}
+
+/**
+We can create an access key using cloudformation, however this results the secret being logged.
```

Comment:
when is the secret logged? When the CloudFormation template gets applied?

Reply:
So basically if we use something like `Fn::GetAtt": [ "MyAccessKey", "SecretAccessKey" ]` it will get logged on the CloudFront logs IIRC. Apparently it was suggested not to use this approach to get secret key values (ref: https://forums.aws.amazon.com/thread.jspa?threadID=266209&tstart=0) thus using this method.

Reply:
Thanks for clarification!